### PR TITLE
[Tidy] Update Vizro-AI error message and safeguards

### PR DESCRIPTION
**Pull request by nadijagraca**
## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

    - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
    - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
    - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
    - I have not referenced individuals, products or companies in any commits, directly or indirectly.
    - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-ai/pyproject.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ dependencies = [
   "langchain>=0.1.0, <0.3.0",  # TODO update to pydantic v2 and remove upper bound
   "langgraph>=0.1.2, <0.2.17",  # TODO update to pydantic v2 and remove upper bound (latest version break pydantic v1 compatibility)
   "python-dotenv>=1.0.0",  # TODO decide env var management to see if we need this
-  "vizro>=0.1.23",
+  "vizro=0.1.29",  # Limited to current version of vizro, until the pydantic V2 changes have been merged
```
Comment (maxschulz-COL) [inline vizro-ai/pyproject.toml:24]:
```suggestion
  "vizro<=0.1.29",  # Limited to current version of vizro, until the pydantic V2 changes of the main repo have been into here
```

**Review by maxschulz-COL — APPROVED** [review]
Looks good - did it work better with weaker models now?

**Comment by nadijagraca** [comment]
> Looks good - did it work better with weaker models now?

Yes, it can create dumbbell chart with 4o-mini without problems. :)

**Comment by stichbury** [comment]
>Yes, it can create dumbbell chart with 4o-mini without problems. :)

Yay! Would you like me to test it out with the goodreads project?

File: vizro-ai/pyproject.toml [hunk 2]
```diff
@@ -21,7 +21,7 @@ dependencies = [
   "langchain>=0.1.0, <0.3.0",  # TODO update to pydantic v2 and remove upper bound
   "langgraph>=0.1.2, <0.2.17",  # TODO update to pydantic v2 and remove upper bound (latest version break pydantic v1 compatibility)
   "python-dotenv>=1.0.0",  # TODO decide env var management to see if we need this
-  "vizro>=0.1.23",
+  "vizro<=0.1.29",  # Limited to current version of vizro, until the pydantic V2 changes of the main repo have been into here
```
Comment (petar-qb) [inline vizro-ai/pyproject.toml:24]: @nadijagraca @maxschulz-COL
Is the upper bound here set due to concerns that Pydantic V2 might negatively impact (or even break) the performance of Vizro-AI?

**Review by petar-qb — APPROVED** [review]
After being bookmarked for a while, I finally found the time to read this article, and it's very relevant to this topic: [Bound Version Constraints](https://iscinumpy.dev/post/bound-version-constraints/). I highly recommend reading it!

**Comment by nadijagraca** [comment]
> > Yes, it can create dumbbell chart with 4o-mini without problems. :)
> 
> Yay! Would you like me to test it out with the goodreads project?

I've tested it already with your prompt and the data you sent me. :)

**Pull request merged**